mne/io/pick.py

```diff
@@ -1104,14 +1104,17 @@ def _picks_str_to_idx(info, picks, exclude, with_ref_meg, return_kind,
     # second: match all to channel names
     #
 
-    bad_name = None
+    bad_names = []
     picks_name = list()
     for pick in picks:
         try:
             picks_name.append(info['ch_names'].index(pick))
         except ValueError:
-            bad_name = pick
-            break
+            bad_names.append(pick)
+
+    if bad_names:
+        logger.warning('Please not that some of the channel names were ignored since they are not appearing in the'
+                    ' instance channels list: {}'.format(str(bad_names)))
```

Comment:
```suggestion
        logger.warning(f'Channel(s) {bad_names} could not be picked, because '
                       'they are not present in the info instance.')
```